spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -429,57 +315,55 @@ public void setZkConnectionTimeout(int zkConnectionTimeout) {
 		// The reference point, if not set explicitly is the latest time for anonymous subscriptions and the
 		// earliest time for group subscriptions. This allows the latter to receive messages published before the group
 		// has been created.
-		long referencePoint = this.startOffset != null ?
-				startOffset.getReferencePoint() : (anonymous ? OffsetRequest.LatestTime() : OffsetRequest.EarliestTime());
+		long referencePoint = properties.getStartOffset() != null ?
+				properties.getStartOffset().getReferencePoint() : (anonymous ? OffsetRequest.LatestTime() : OffsetRequest.EarliestTime());
 		return createKafkaConsumer(name, inputChannel, properties, consumerGroup, referencePoint);
 	}
 
+
 	@Override
-	public Binding<MessageChannel> bindProducer(final String name, MessageChannel moduleOutputChannel, Properties properties) {
+	public Binding<MessageChannel> doBindProducer(final String name, MessageChannel moduleOutputChannel, KafkaProducerProperties properties) {
+
 		Assert.isInstanceOf(SubscribableChannel.class, moduleOutputChannel);
-		KafkaPropertiesAccessor producerPropertiesAccessor = new KafkaPropertiesAccessor(properties);
-		validateProducerProperties(name, properties, SUPPORTED_PRODUCER_PROPERTIES);
 		if (logger.isInfoEnabled()) {
 			logger.info("Using kafka topic for outbound: " + name);
 		}
 
 		validateTopicName(name);
 
-		int numPartitions = producerPropertiesAccessor.getNumberOfKafkaPartitionsForProducer();
+		int partitionCount = properties.getPartitionCount();
+		if (partitionCount == 0) {
+			throw new IllegalArgumentException("Module count cannot be zero");
```

Comment:
I'm wondering if we should remove this test here and put it on the bean - maybe with some JSR-303 validation on bind.  It's sort of awkward that we test it only for Kafka (it can't be zero in general).